src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -228,118 +221,177 @@ private void SetText(ref string text)
-        {
-            color = AppColor.AnsiTerminalRedBackBold.GetThemeColor();
-            markerStrategy.AddMarker(new TextMarker(lineRemoved.Offset + beginOffset,
-                                                    lineRemoved.Length - beginOffset - reverseOffset,
-                                                    TextMarkerType.SolidBlock, color,
-                                                    ColorHelper.GetForeColorForBackColor(color)));
-        }
+        TextMarker CreateDimmedMarker(ISegment line, int offset, int length, Color color)
+            => CreateTextMarker(line.Offset + beginOffset + offset, length, ColorHelper.DimColor(ColorHelper.DimColor(color)));
+
+        static TextMarker CreateTextMarker(int offset, int length, Color color)
+            => new(offset, length, TextMarkerType.SolidBlock, color, ColorHelper.GetForeColorForBackColor(color));
+
+        static Color GetAddedBackColor() => AppColor.AnsiTerminalGreenBackNormal.GetThemeColor();
+        static Color GetAddedForeColor() => AppColor.AnsiTerminalGreenForeBold.GetThemeColor();
+        static Color GetRemovedBackColor() => AppColor.AnsiTerminalRedBackNormal.GetThemeColor();
+        static Color GetRemovedForeColor() => AppColor.AnsiTerminalRedForeBold.GetThemeColor();
     }
 
-    private void AddExtraPatchHighlighting(IDocument document)
+    /// <summary>
+    ///  Matches related removed and added lines in a consecutive block of a patch document and marks identical parts dimmed.
+    /// </summary>
+    private void MarkInlineDifferences(IDocument document)
     {
         int line = 0;
-
         bool found = false;
-        int diffContentOffset;
-        List<ISegment> linesRemoved = GetRemovedLines(document, ref line, ref found);
-        List<ISegment> linesAdded = GetAddedLines(document, ref line, ref found);
-        if (linesAdded.Count == 1 && linesRemoved.Count == 1)
-        {
-            ISegment lineA = linesRemoved[0];
-            ISegment lineB = linesAdded[0];
-            if (lineA.Length > 4 && lineB.Length > 4 &&
-                document.GetCharAt(lineA.Offset + 4) == 'a' &&
-                document.GetCharAt(lineB.Offset + 4) == 'b')
-            {
-                diffContentOffset = 5;
-            }
-            else
-            {
-                diffContentOffset = 4;
-            }
 
-            MarkDifference(document, linesRemoved, linesAdded, diffContentOffset);
+        // Skip the first pair of removed / added lines, which uses to contain the filenames - but without highlighting
+        _ = GetRemovedLines(document, ref line, ref found);
+        if (found)
+        {
+            _ = GetAddedLines(document, ref line, ref found);
         }
 
-        // overlap when marking
-        diffContentOffset = 1;
+        // Process the next blocks of removed / added lines and mark in-line differences
+        const int diffContentOffset = 1; // in order to skip the prefixes '-' / '+'
```

Comment:
Note: This is incorrect for CombinedDiff.
The prefix length could be derived from the highlighter, that will improve inline marking.
However, I think we just can ignore this...

![image](https://github.com/user-attachments/assets/b6b00ade-b3a5-4a33-8e96-3787164145d7)

Reply:
I have not changed this, i.e. separate issue.